src/legacy/server/saved_objects/service/saved_objects_client.ts

```diff
@@ -99,7 +99,7 @@ export interface MigrationVersion {
 }
 
 export interface SavedObjectAttributes {
-  [key: string]: SavedObjectAttributes | string | number | boolean | null;
+  [key: string]: SavedObjectAttributes | string | number | boolean | null | undefined;
```

Comment:
@elastic/kibana-platform @skaapgif is there any reason we shouldn't add `undefined` to this type? Some of the fields are optional in our [new SavedQuery type](https://github.com/elastic/kibana/blob/0157959a3c011cb63c0e9a11ee0aa78d28343ece/src/legacy/core_plugins/data/public/search/search_bar/index.tsx#L33-L47) and we'd rather not pass them as null if possible.

Reply:
👍 `undefined` was also added in https://github.com/elastic/kibana/pull/37042 so it might land in master soon.